src/Hl7.Fhir.Specification/Specification/Source/Summary/ArtifactSummaryDetails.cs

```diff
@@ -0,0 +1,200 @@
+﻿using Hl7.Fhir.ElementModel;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Utility;
+using System;
+using System.Linq;
+
+namespace Hl7.Fhir.Specification.Source.Summary
+{
+    // DSTU2 Specific!
+
+    /// <summary>For accessing common artifact summary details.</summary>
+    /// <remarks>Common details are initialized by the <see cref="ArtifactSummaryGenerator"/> itself.</remarks>
+    public static class ArtifactSummaryDetails
```

Comment:
Doesn't this mean that all extension methods on IArtifactSummaryDetailsProvider are in scope all the time?  So, now there's no real notion of "these attributes are right for a ValueSet"?

Reply:
Since this are all extension methods on static classes, why do we still have an ArtifactSummaryFactory? There are no specific classes of instances to create?

Reply:
1) Correct, I removed all concrete subclasses to simplify the default API design. As a consequence, we lose some type safety.
2) Client can override factor to create specific subclasses with additional type-safe properties. However by default, the API generates standard ArtifactSummary instances.

Reply:
Since we've chosen to have no subclasses, I think we should ditch the factory pattern for ArtifactSummaries then.  Your namespace will contain those extension methods, so for the API user this will only create confusion.